### ISSUE TITLE
bytestring+binary

### ISSUE THREAD

**Issue by bergmark** [haskell-infra/hackage-trustees#91]
A couple of build failures, tested with GHC-7.8.4:

```
V~/D/binary> cabal install --constraint 'bytestring==0.10.2.0' binary-0.8.4.0 -j1
Resolving dependencies...
Downloading binary-0.8.4.0...
cabal: Entering directory '/var/folders/sd/d9hgvrbx4dggjz1bxkpw5qgc0000gn/T/cabal-tmp-21629/binary-0.8.4.0'
Configuring binary-0.8.4.0...
Building binary-0.8.4.0...
Preprocessing library binary-0.8.4.0...

src/Data/Binary/Builder.hs:71:18:
    Could not find module ‘Data.ByteString.Short’
    It is a member of the hidden package ‘bytestring-0.10.4.0’.
    Perhaps you need to add ‘bytestring’ to the build-depends in your .cabal file.
    Use -v to see a list of the files searched for.
cabal: Leaving directory '/var/folders/sd/d9hgvrbx4dggjz1bxkpw5qgc0000gn/T/cabal-tmp-21629/binary-0.8.4.0'
```

```
V~/D/binary> cabal install --constraint 'bytestring==0.10.2.0' binary-0.8.4.1 -j1
Resolving dependencies...
cabal: Entering directory '/var/folders/sd/d9hgvrbx4dggjz1bxkpw5qgc0000gn/T/cabal-tmp-21723/binary-0.8.4.1'
Configuring binary-0.8.4.1...
Building binary-0.8.4.1...
Preprocessing library binary-0.8.4.1...
[1 of 9] Compiling Data.Binary.Internal ( src/Data/Binary/Internal.hs, dist/dist-sandbox-3c5a6a57/build/Data/Binary/Internal.o )
[2 of 9] Compiling Data.Binary.Get.Internal ( src/Data/Binary/Get/Internal.hs, dist/dist-sandbox-3c5a6a57/build/Data/Binary/Get/Internal.o )
[3 of 9] Compiling Data.Binary.FloatCast ( src/Data/Binary/FloatCast.hs, dist/dist-sandbox-3c5a6a57/build/Data/Binary/FloatCast.o )
[4 of 9] Compiling Data.Binary.Builder ( src/Data/Binary/Builder.hs, dist/dist-sandbox-3c5a6a57/build/Data/Binary/Builder.o )

src/Data/Binary/Builder.hs:76:1:
    Data.ByteString.Builder: Can't be safely imported!
    The module itself isn't safe.

src/Data/Binary/Builder.hs:77:1:
    Data.ByteString.Builder.Prim: Can't be safely imported!
    The module itself isn't safe.

src/Data/Binary/Builder.hs:79:1:
    Data.ByteString.Builder.Extra: Can't be safely imported!
    The module itself isn't safe.
cabal: Leaving directory '/var/folders/sd/d9hgvrbx4dggjz1bxkpw5qgc0000gn/T/cabal-tmp-21723/binary-0.8.4.1'
Failed to install binary-0.8.4.1
cabal: Error: some packages failed to install:
binary-0.8.4.1 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by sol** [comment]
Affected versions of `binary` are `0.8.3.0`, `0.8.4.0` and `0.8.4.1`.  Can somebody please add `bytestring >=0.10.4` to these? (@phadej).

**Comment by phadej** [comment]
@bergmark was there some concerns why you didn't add the bounds?

**Comment by bergmark** [comment]
I didn't get the time to figure out what the correct change was

**Comment by phadej** [comment]
@bergmark I'll check on Thursday or Friday if nobody does that before. A bit busy atm myself too.

**Comment by kolmodin** [comment]
I fixed kolmodin/binary#123 for future versions of `binary`. Bumped the versions on hackage too.

**Comment by bergmark** [comment]
Looks like @kolmodin sorted this out, thanks!
